llama_index/schema.py

```diff
@@ -48,11 +47,10 @@ class BaseDocument(DataClassJsonMixin):
 
     def __post_init__(self) -> None:
         """Post init."""
-        # assign doc_id if not set
-        if self.doc_id is None:
-            self.doc_id = get_new_id(set())
         if self.doc_hash is None:
             self.doc_hash = self._generate_doc_hash()
+        if self.doc_id is None:
+            self.doc_id = self.doc_hash
```

Comment:
Relying on this behavior would be surprising (especially because user can modify doc_id)